File: C++/Labs/Lab8C.cpp
#include <iostream>

int main() {
  int rows;
  int columns;

  std::cout << "Enter the number of rows: ";
  std::cin >> rows;
  std::cout << "Enter the number of columns: ";
  std::cin >> columns;
  int array1[rows][columns];

  std::cout << std::endl << "I have " << rows << " rows and " << columns << " columns. I need to fill up " << rows*columns << " spaces." << std::endl;
  std::cout << std::endl << "The " << rows << "x" << columns << " 2-D array: " << std::endl;
  int row = 0;
  int column = 0;

  for (int i = 0; i < rows*columns; i++) {
    if (column == columns) {
      column = 0;
      row++;
      std::cout << std::endl;
    }
    array1[row][column] = i+1;
    std::cout << array1[row][column] << "| ";
    column++;
  }

  int array2[rows*columns];

  row = 0;
  column = 0;
  std::cout << std::endl << std::endl << "The " << rows << "x" << columns << " 2-D array flattened into a 1-D array: " << std::endl;

  for (int i = 0; i < rows*columns; i++) {

    if (column == columns) {
      column = 0;
      row++;
    }

    array2[i] = array1[row][column];
    std::cout << array2[i] << "| ";
    column++;
  }

}

File: Java/Lab8C.java
import java.util.Scanner;

public class Lab8C {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        System.out.print("Please enter the number of rows: ");
        int rows = scan.nextInt();
        System.out.print("Please enter the number of columns: ");
        int columns = scan.nextInt();

        int array[][] = new int[rows][columns];

        System.out.println("\nI have " + rows + " rows and " + columns + " columns. I need to fill-up " + rows*columns + " spaces.\n");
        System.out.println("The " + rows + "x" + columns + " array:");

        int rowCount = 0;
        int columnCount = 0;

        for (int i = 0; i < rows*columns; i++) {
            array[rowCount][columnCount] = (i + 1);
            columnCount++;
            if (columnCount == (columns)) {
                columnCount = 0;
                rowCount++;

            }
            if (rowCount == (rows)) {
                rowCount = 0;
                columnCount = 0;

            }
        }
        for (int i = 0; i < rows*columns; i++) {
            System.out.print(array[rowCount][columnCount] + "|");
            columnCount++;
            if (columnCount == (columns)) {
                columnCount = 0;
                rowCount++;
                System.out.println("");
            }
            if (rowCount == (rows)) {
                rowCount = 0;
                columnCount = 0;
            }
        }
        System.out.println("\nThe " + rows + "x" + columns + "1 array flattened into a " + rows*columns + " cell 2-D array:");

        int array1d[] = new int[rows*columns];

        for (int i = 0; i < (rows*columns); i++) {
            array1d[i] = array[rowCount][columnCount];
            columnCount++;
            if (columnCount == columns) {
                columnCount = 0;
                rowCount++;
            }
            if (rowCount == (rows)) {
                rowCount = 0;
                columnCount = 0;
            }
        }
        for (int i = 0; i < rows*columns; i++) {
            System.out.print(array1d[i] + "|");
        }

    }
}

